internal/commands/pull/inputs_test.go

```diff
@@ -159,3 +160,169 @@ func TestPullInputsResolveRemoteApp(t *testing.T) {
+		})
+
+		t.Run("should return an error if the template ID is not compatible with the app", func(t *testing.T) {
+			var realmClient mock.RealmClient
+
+			realmClient.CompatibleTemplatesFn = func(groupID, appID string) ([]realm.Template, bool, error) {
+				return []realm.Template{{ID: "some-template-id", Name: "some template name"}}, true, nil
+			}
+
+			_, ui := mock.NewUI()
+			input := inputs{TemplateID: "wrong-template-id"}
+			result, err := input.resolveClient(ui, realmClient, "some-group-id", "some-app-id")
+			assert.Nil(t, result)
+			assert.Equal(t, "template wrong-template-id is not compatible with this app", err.Error())
+		})
+
+		t.Run("should return the template client if it is compatible with this app", func(t *testing.T) {
+			var realmClient mock.RealmClient
+
+			realmClient.CompatibleTemplatesFn = func(groupID, appID string) ([]realm.Template, bool, error) {
+				return []realm.Template{{ID: "some-template-id", Name: "some name"}, {ID: "some-other-template-id", Name: "some other name"}}, true, nil
+			}
+			realmClient.ClientTemplateFn = func(groupID, appID, templateID string) (*zip.Reader, error) {
+				return &templateZipPkg1.Reader, nil
+			}
+
+			_, ui := mock.NewUI()
+			input := inputs{TemplateID: "some-template-id"}
+			result, err := input.resolveClient(ui, realmClient, "some-group-id", "some-app")
+			assert.Nil(t, err)
+			assert.Equal(t, 1, len(result))
+			compareZipPackages(t, templateZipPkg1, result["some-template-id"])
+		})
+	})
+	t.Run("without a template ID passed in", func(t *testing.T) {
+		input := inputs{}
+		t.Run("should return nothing if the app is not made with a template", func(t *testing.T) {
+			var realmClient mock.RealmClient
+
+			realmClient.CompatibleTemplatesFn = func(groupID, appID string) ([]realm.Template, bool, error) {
+				return nil, false, nil
+			}
+			_, ui := mock.NewUI()
+			result, err := input.resolveClient(ui, realmClient, "some-group-id", "some-app-id")
+			assert.Nil(t, result)
+			assert.Nil(t, err)
+		})
+		t.Run("should return nothing if the user refuses to export with compatible templates", func(t *testing.T) {
+			var realmClient mock.RealmClient
+			realmClient.CompatibleTemplatesFn = func(groupID, appID string) ([]realm.Template, bool, error) {
+				return []realm.Template{{ID: "some-template-id", Name: "some name"}}, true, nil
+			}
+			_, console, _, ui, err := mock.NewVT10XConsole()
+			assert.Nil(t, err)
+			defer console.Close()
+
+			doneCh := make(chan struct{})
+			go func() {
+				defer close(doneCh)
+				console.ExpectString("Would you like to export with a template?")
+				console.SendLine("n")
+				console.ExpectEOF()
+			}()
+
+			result, err := input.resolveClient(ui, realmClient, "some-group-id", "some-group-app")
+			assert.Nil(t, result)
+			assert.Nil(t, err)
+		})
+
+		t.Run("should prompt the user to export with compatible templates and return the templates selected", func(t *testing.T) {
```

Comment:
it seems that this test (and potentially others) are already covered in some of your `pull_test` command Handler tests.  if that is the case, let's remove the duplicate ones from this file

Reply:
I'm iffy on removing the tests even if they duplicate in the areas that they cover because the PullHandler tests are concerned with how `resolveClientTemplates` works in regards to the pull function, but these tests are for the different cases that _can_ happen in `resolveClientTemplates`.

Reply:
definitely don't remove coverage, but in particular it seems to me this is specifically covered with the `"should export all selected compatible templates if no template id is passed in and the app is made with a template"` test (verifying the contents an all).  at the very least, we might be able to remove this one (and the `compareZipPkg` helper method along with it)